src/ecore.js

```diff
@@ -861,6 +868,9 @@ EClass.values = {
 
         return _.isArray(subTypes) ? subTypes : [];
     },
+    hasSubTypes: function () {
+      return this.get('abstract') ||  this.get('eAllSubTypes');
```

Comment:
Agreed on adding the `this.get('eAllSubTypes').length`. We struggled with what to call this function too.  Do you have any suggestions for the name instead of hasSubTypes? Some potential names are couldHaveSubType, potentiallyHasSubType, or checkForSubType.